src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1603,6 +1603,27 @@ private void ValidateAssignment(
                         }
                     }
                 }
+                else
+                {
+                    switch (op1)
+                    {
+                        case BoundPropertyAccess { PropertySymbol.SetMethod: { } propSet, ReceiverOpt: var receiver } when propSet.GetIsNewExtensionMember():
+                            var methodInvocationInfo = new MethodInvocationInfo
```

Comment:
Consider moving this construction into a `From` factory. I think there is value in seeing all of the construction cases in a single location. In the future if we add more fields makes it easier to evaluate the impact.